### ISSUE TITLE
Something is interfering on the tests

### ISSUE THREAD

**Issue by nei** [tschaub/mock-fs#29]
Hi everyone, I have this test case that works when I run just mocha tests but fail when I run on my default task with others tasks. Does anyone know what can be interfering on this? Any suggestion?

```
/*jshint expr: true*/
require('chai').should();
var proxyquire = require('proxyquire').noCallThru().noPreserveCache();
var sinon = require('sinon');
var mockFs = require('mock-fs');
var path = require('path');
var req = {};

describe.only('Components rendering', function(){

    beforeEach(function (done) {
        req = {
            project: {
                brand: 'bla',
                page: 'home',
                locale: 'en-gb',
                domain: 'bla.com'
            },
            query: {}
        };

        done();
    });

   afterEach(function (done){
        mockFs.restore();
        done();
    });

    it('should extend page config properly with component file', function (done) {

        var homeConfig = JSON.stringify({
            type: 'homepageLayout',
            slots: {
                'slot1': 'value1',
                '<<aComponent': {
                     file: 'theComponent.json'
                },
                '>>aSlot': {
                    file: 'header.json'
                }
            }
        });
        var componentJson = JSON.stringify({
            fooComponent: 'barComponent'
        });
        var headerJson = JSON.stringify({
            foo: 'bar'
        });

        var fsToMock = {};
        var pathFile = path.join(__dirname, '/../../../', process.env.PROJECT_ROOT, '/pageconfig/bla.com/en-gb/');
        var pathFile2 = path.join(__dirname, '/../../../', process.env.PROJECT_ROOT, '/pageconfig/project.com/en-gb/');

        fsToMock[path.join(pathFile + 'home.json')] = homeConfig;
        fsToMock[path.join(pathFile2 + 'home.json')] = homeConfig;

        fsToMock[path.join(pathFile + 'common/header.json')] = headerJson;
        fsToMock[path.join(pathFile + 'component/theComponent.json')] = componentJson;

        fsToMock[path.join(pathFile2 + 'common/header.json')] = headerJson;
        fsToMock[path.join(pathFile2 + 'component/theComponent.json')] = componentJson;

        var getConfig = require('../../../app/lib/page/getConfig');
        mockFs(fsToMock);

        getConfig(req, {}, function () {
            mockFs.restore();
            var expected = {
                slot1: 'value1',
                aComponent: {
                    fooComponent: 'barComponent'
                },
                aSlot: {
                    foo: 'bar'
                }
            };
            req.project.pageConfig.slots.should.deep.equal(expected);
            done();
        });

    });
});
```

The error is basically because it is not finding the file.

```
ENOENT, no such file or directory '/home/vagrant/project/app/pageconfig/project.com/en-gb/home.json'

and then

TypeError: Cannot read property 'slots' of undefined
      at /home/vagrant/project/test/server/lib/getConfigComponents.js:78:37
      at module.exports (/home/vagrant/project/app/lib/page/getConfig.js:69:20)
      at Context.<anonymous> (/home/vagrant/project/test/server/lib/getConfigComponents.js:67:9)
      at Test.Runnable.run (/home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runnable.js:204:15)
      at Runner.runTest (/home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runner.js:374:10)
      at /home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runner.js:452:12
      at next (/home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runner.js:299:14)
      at /home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runner.js:309:7
      at next (/home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runner.js:247:23)
      at /home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runner.js:271:7
      at done (/home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runnable.js:195:5)
      at /home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runnable.js:207:9
      at Context.<anonymous> (/home/vagrant/project/test/server/lib/getConfigComponents.js:22:9)
      at Hook.Runnable.run (/home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runnable.js:204:15)
      at next (/home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runner.js:259:10)
      at Object._onImmediate (/home/vagrant/project/node_modules/grunt-mocha-test/node_modules/mocha/lib/runner.js:276:5)
```

My default task which this test is failing is like this:
```
grunt.registerTask('default', [
        'clean:coverage',
        'lint',
        'compile-assets',
        'coverageandtest',
        'clienttest',
        'security'
    ]);
```

and the one I use and it pass is like this:
```
    grunt.registerTask('servertest', ['env:test', 'mochaTest:test', 'env:notTest']);
```

**Comment by Rleahy22** [comment]
:+1:

**Comment by Rleahy22** [comment]
+1

**Comment by grantcarthew** [comment]
+1

**Comment by tschaub** [comment]
Thanks for the additional detail.  I'll close this for now, but please reopen or add more detail if this can be reproduced in a way that suggests an issue with `mock-fs`.

**Comment by grantcarthew** [comment]
There are more details @tschaub. I added to my test files and found the error returned. I suspect it is because the test files were running in parallel in some way causing the mock file system to prevent other tests from running (unable to require modules).

I came up with a fix though. Knowing that the node `require` call is synchronous, I created the following [test-runner.js](https://github.com/grantcarthew/node-scalable-blob-store/blob/master/tests/test-runner.js) file:

```js
require('./options-parser.spec')
require('./id-generator.spec')
require('./id-validator.spec')
require('./fs-blob-item-list.spec')
require('./fs-blob-dir-latest.spec')
require('./fs-blob-dir-latest-full-depth.spec')
require('./blob-path-build.spec')
require('./blob-store-api.spec')
require('./blob-store-multi.spec')
require('./blob-store-bulk.spec')
```

So rather than call `tape ./tests/**/*.spec.js` I am calling `node ./tests/test-runner.js | tap-spec`.
See my [package.json](https://github.com/grantcarthew/node-scalable-blob-store/blob/master/package.json) for the details.

Notice I have added piping to tap-spec again. This test-runner.js seems to have fixed the issue. The tests have not failed since.

This might work for you also @nei.

Hope this helps.
